src/clp_ffi_js/ir/LogEventWithFilterData.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+#define CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+
+#include <utility>
+
+#include <clp/ir/types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+/**
+ * A class which accepts a log event type as a template parameter and provides additional members
+ * for the log level and timestamp. The additional members facilitate log level filtering.
+ * @tparam LogEvent The type of the log event.
+ */
+template <typename LogEvent>
```

Comment:
If I'm not wrong, all template parameters should following naming convention `type_name_t`.
cc @LinZhihao-723 / @kirkrodrigues please help confirm.

Reply:
Only primitive template parameters are named following the format type_name_t. Non-primitives are named how Marco has them.

Reply:
A little adding on:
Since we're using C++20, we should use `requires` or define a concept about the accepted LogEvent type

Reply:
@davemarco caution, i could be mistaken but i remember having issues with Emscripten + Ninja + C++20 features when we initialize this project, and not finding the `concepts` headers were one of the symptoms. The mitigation to use Unix Makefiles was removed in #19 . You may try adding the mitigation back if you see odd issues compiling the code.

Reply:
Although the types that this class will accept don't need to conform to any interface, the description I suggested does indicate that they need to be log events with fields. Not all types (e.g. `int64_t`) would fit that description (unless we're using a very loose definition), so in that case, using a concept/requires statement to restrict the types to the existing LogEvent types would be beneficial.

Let me know if you disagree.